app/controllers/lti_controller.rb

```diff
@@ -0,0 +1,81 @@
+class LtiController < ApplicationController
+  skip_before_action :verify_authenticity_token, only: :launch # for lti integration
+  before_action :set_group_assignment, only: %i[launch]
+  before_action :set_lti_params, only: %i[launch]
+  after_action :allow_iframe_lti, only: %i[launch]
+  
+  def launch
+    session[:isLTI]=true # the lti session starting
+    require 'oauth/request_proxy/action_controller_request'
+
+    if @assignment.blank?
+      # if no assignment is found
+      flash[:notice] = t("lti.launch.notice_no_assignment")
+      render :launch_error, status: 401
+      return
+    end
+    
+    if @group.present? # if there is a valid group based for the lti_token_key
+      @provider = IMS::LTI::ToolProvider.new(
+        params[:oauth_consumer_key], # lms_oauth_consumer_key
+        @assignment.lti_shared_secret, # the group's lti_token
+        params
+      )
+
+      if !@provider.valid_request?(request) # checking the lti request from the lms end
+        render :launch_error, status: 401
+        return
+      end
+
+      @user = User.find_by(email: @email_from_lms) # find user by matching email with circuitverse and lms 
+
+      if @user.present? # user is present in cv 
+        if @user.id == @group.mentor_id # user is teacher
+          sign_in(@user) # passwordless sign_in the user as the authenticity is verified via lms
+          lms_auth_success_notice = t("lti.launch.notice_lms_auth_success_teacher", email_from_lms: @email_from_lms, lms_type: @lms_type, course_title_from_lms: @course_title_from_lms)
+          redirect_to group_assignment_path(@group, @assignment), notice: lms_auth_success_notice # if auth_success send to group page
+        else
+          user_in_group = GroupMember.find_by(user_id:@user.id,group_id:@group.id) # check if the user belongs to the cv group
+
+          if user_in_group.present? # user is member of the group
```

Comment:
should be something like @group.members.exists?(user: @user)

Reply:
This error is happening
<img width="1108" alt="image" src="https://user-images.githubusercontent.com/52851184/123685840-bb936a00-d86c-11eb-8743-f90eea041780.png">